src/main.js

```diff
@@ -269,12 +275,14 @@ module.exports = {
         const text = textEditor.getText()
 
         let rules = {}
-        if (textEditor.isModified() && Object.keys(ignoredRulesWhenModified).length > 0) {
-          rules = ignoredRulesWhenModified
-        }
-        if (textEditor.isModified() && ignoreFixableRulesWhileTyping) {
-          // Note that this list will only contain rules after the first lint job
-          rules = idsToIgnoredRules(helpers.rules.getFixableRules())
+        if (textEditor.isModified()) {
+          const ignoredRules = new Set()
```

Comment:
Is there any way to avoid creating a new set and looping through all the `ignoredRulesWhenModified` and fixable rules each time the editor is modified?

Reply:
The fixable rules could change at any time, so there isn't _too_ much point in saving the generated list.

It's certainly possible though.

Reply:
What if we use `if (textEditor.isModified() && ignoreFixableRulesWhileTyping)` at least?

Reply:
That's essentially what's already happening?

```js
if (textEditor.isModified()) {
  // ...
  if (ignoreFixableRulesWhileTyping) {
```

Reply:
By moving that check up top, if `ignoreFixableRulesWhileTyping` is false, we don't need to create a new `ignoredRules` set and move all of the `ignoredRulesWhenModified` into it.

Reply:
Once we have this in a data-store, we can use `reselect` to memoize a selector on this data.

Reply:
Handling of both `ignoreFixableRulesWhileTyping` and `ignoredRulesWhenModified` is gated behind `textEditor.isModified()`, so in this I just moved them both under a single check since there is no reason to check that twice.

Is this what you mean?
```js
if (textEditor.isModified()) {
  rules = idsToIgnoredRules(ignoredRulesWhenModified)
  if (ignoreFixableRulesWhileTyping) {
    // Note that this list will only contain rules after the first lint job
    rules = Object.assign(rules, idsToIgnoredRules(helpers.rules.getFixableRules())
  }
}
```